flink-table/flink-table-planner/src/test/java/org/apache/flink/table/planner/calcite/FlinkCalciteSqlValidatorTest.java

```diff
@@ -109,7 +109,20 @@ void testInvalidNumberOfColumnsWhileInsertInto(String sql) {
                 "INSERT INTO t2 (a, b) WITH cte AS (SELECT 1, 2) SELECT * FROM cte",
                 "INSERT INTO t2 (a, b) WITH cte AS (SELECT * FROM t2_copy) SELECT * FROM cte",
                 "INSERT INTO t2 (a, b) WITH cte AS (SELECT t1.a, t2_copy.b FROM t1, t2_copy) SELECT * FROM cte",
-                "INSERT INTO t2 (a, b) WITH cte1 AS (SELECT 1), cte2 AS (SELECT 2) SELECT * FROM cte1, cte2"
+                "INSERT INTO t2 (a, b) WITH cte1 AS (SELECT 1), cte2 AS (SELECT 2) SELECT * FROM cte1, cte2",
+                "INSERT INTO t2 (a, b) "
+                        + "WITH cte1 AS (SELECT 1, 2), "
+                        + "cte2 AS (SELECT 2, 3) "
+                        + "SELECT * FROM cte1 UNION SELECT * FROM cte2",
```

Comment:
There are failing unit tests with CTEs above.  Do we want some UNIONs which do not work above?  Definitely a non-blocking suggestion.

Reply:
With latest commit all tests passed
so shouldn't be an issue I guess

Reply:
Sounds good!